Akinator.Api.Net/Utils/GuessDueChecker.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Akinator.Api.Net.Model;
+
+namespace Akinator.Api.Net.Utils
+{
+    public static class GuessDueChecker
+    {
+        public static bool GuessIsDue(AkinatorQuestion currentQuestion, int lastGuessTakenAtStep)
+        {
+            if (currentQuestion is null)
+            {
+                return false;
+            }
+
+            var stepsTakenSinceLastGuess = currentQuestion.Step - lastGuessTakenAtStep;
+            if (NoMoreQuestions() ||
+                currentQuestion.Step >= 80)
+            {
+                return true;
+            }
+
+            if (stepsTakenSinceLastGuess < 5)
+            {
+                return false;
+            }
+
+            if (currentQuestion.Step <= 25)
+            {
+                return !(currentQuestion.Progression <= 97.0f);
+            }
+
+            if (currentQuestion.Progression <= 80.0f && stepsTakenSinceLastGuess < 30)
+            {
+                return false;
+            }
+
+            return 80 - currentQuestion.Step > 5;
+        }
```

Comment:
@janniksam I think the logic that I added in my latest commit was better that this (Android) one, I have tested it and got better results with the old (my latest commit) than this (Android) one

Reply:
I would like to keep it as original as possible. Hope you understand.

Reply:
ok, as you like :) it was just my opinion